lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,17 @@ def word_symbol_pair?(pair)
           return false unless key.sym_type?
 
           sym_name = key.loc.expression.source
-          sym_name !~ /\A:["']|=\z/
+          return false if ends_with_equals? sym_name
+          valid_19_syntax_symbol? sym_name
+        end
+
+        def valid_19_syntax_symbol?(sym_name)
+          sym_name.sub!(/\A:/, '')
+          RuboCop::ProcessedSource.new("{ #{sym_name}: :foo }").valid_syntax?
+        end
+
+        def ends_with_equals?(sym_name)
```

Comment:
Yeah, clearly I was not paying attention. I'll just fold that into the valid syntax method, but right now it does seem to be needed, at least for the quote checking it does.